src/main/scala/org/constellation/p2p/Cluster.scala

```diff
@@ -145,6 +146,21 @@ class Cluster[F[_]](
   dao.metrics.updateMetricAsync[IO]("nodeState", initialState.toString).unsafeRunAsync(_ => ())
   private val stakingAmount = ConfigUtil.getOrElse("constellation.staking-amount", 0L)
 
+  def createAndSetNewOwnToken: F[Token] =
```

Comment:
1. why not passing `SessionTokenService` as a dependency there?
2. why mimic all methods in cluster?

Reply:
SessionTokenService is in dao so I didn't put it in constructor. Regarding mimicing the functions you are right, there is no need to do that, I probably created one wrapper to have a function on F and than followed with other functions in the same manner. But I can either Lift to F at the spot where I need it, or even simpler have sessionTokenService[F] in constructor.

Reply:
Every service is in `dao`. We don't want to pass everything implicitly by `dao`, as we're trying to limit implicit dao usages and remove the current dao at all.

Reply:
Passing it via constructor as an explicit dependency would be enough and will save you from lifting to F